include/dqlite.h

```diff
@@ -17,8 +17,304 @@
+
+/**
+ * Indicate that this server will be the "bootstrap" server for its cluster.
+ *
+ * The bootstrap server should be the first to start up. It automatically
+ * becomes the leader in the first term, and is responsible for adding all other
+ * servers to the cluster configuration. There must be exactly one bootstrap
+ * server in each cluster. After the first startup, the bootstrap server is no
+ * longer special and this function is a no-op.
+ */
+int dqlite_server_set_bootstrap(dqlite_server *server);
+
+/**
+ * Declare the address of an existing server in the cluster, which should
+ * already be running.
+ *
+ * This function
+ * multiple times with different addresses. The server addresses declared with
+ * this function will not be used unless @server is starting up for the first
+ * time; after the first startup, the list of servers stored on disk will be
+ * used instead. (It is harmless to call this function unconditionally.)
+ */
+int dqlite_server_set_peer_address(dqlite_server *server, const char *addr);
+
+/**
+ * Configure @server to listen on the address @addr for incoming connections
+ * (from clients and other servers).
+ *
+ * If no bind address is configured with this function, the abstract address
+ * passed to dqlite_server_create will be used. The point of this function is to
+ * support decoupling the abstract address from the networking implementation
+ * (for example, if a proxy is going to be used).
+ *
+ * @addr must use one of the following formats:
+ *
+ * 1. "<HOST>"
+ * 2. "<HOST>:<PORT>"
+ * 3. "@<PATH>"
+ *
+ * Where <HOST> is a numeric IPv4/IPv6 address, <PORT> is a port number, and
+ * <PATH> is an abstract Unix socket path. The port number defaults to 8080 if
+ * not specified. In the second form, if <HOST> is an IPv6 address, it must be
+ * enclosed in square brackets "[]". In the third form, if <PATH> is empty, the
+ * implementation will automatically select an available abstract Unix socket
+ * path.
+ *
+ * If an abstract Unix socket is used, the server will accept only
+ * connections originating from the same process.
+ */
+int dqlite_server_set_bind_address(dqlite_server *server, const char *addr);
+
+/**
+ * Configure the function that this server will use to connect to other servers.
+ *
+ * The same function will be used by the server's attached client to establish
+ * connections to all servers in the cluster. @arg is a user data parameter that
+ * will be passed to all invocations of the connect function.
+ */
+int dqlite_server_set_connect_func(dqlite_server *server,
+				   dqlite_connect_func f,
+				   void *arg);
+
+/**
+ * Configure the target number of voting nodes for this server's cluster.
+ *
+ * When this server is the cluster leader, it will try to promote and demote
+ * other servers to reach the target voter count. You should configure the same
+ * target on all servers that will participate in the cluster.
+ */
+int dqlite_server_set_target_voters(dqlite_server *server, int n);
```

Comment:
The fact that we require to configure the same value on all servers is a bit unfortunate. I believe we have a few cluster-wide settings (target voters, target standbys and network latency come to my mind) that we currently configure/persist on a per-server basis, and could instead be saved either in the raft log or (better) in a dqlite table or db. In that way we would not require users to book-keep these settings and to make sure they are in sync across all servers.

For example, setting or changing the number of target voters could be done by running a bit of SQL, like:

```
PRAGMA voters=5;
```

or `PRAGMA dqlite.voters=5;` or `UPDATE dqlite.config SET value=5 WHERE key='voters'` or whatever other SQL.

See also https://www.sqlite.org/pragma.html.

This will probably be more work to implement, though.

If it's critical to expose these settings in the very first iteration of this C client support, it'd be also okay to leave `dqlite_server_set_target_voters` and `dqlite_server_set_target_standbys` as they are in this PR. In the future we could implement the more convenient cluster-wide SQL-based setting, and make it take precedence if used.

If it's not so critical to have these settings exposed in the very first iteration of the C client (after the default settings should be good enough for most use cases), maybe we could not implement these functions right now, and expose these settings later via SQL.